### ISSUE TITLE
timesearch.py - get_all_posts(), lower=upper when too few results are found

### ISSUE THREAD

**Issue by rroyo** [voussoir/reddit#11]
Hello,

first of all, thank you for sharing your code, it has helped me a lot to build my own reddit bot.

https://github.com/voussoir/reddit/blob/master/Prawtimestamps/timesearch.py#L307

Now on the issue I've found, I've taken the get_all_posts() function to use it in my own code and I've noticed that when too few results are found, the lower variable gets assigned the value of the upper, when it should stay the same and recheck with a higher upper limit.

If the first if is true, then the second is false and the code inside else gets executed, changing the value of lower, when it should stay the same.

```
if itemsfound < 75:
    ...
if itemsfound > 99:
    ...
else:
    lower = upper
    ...
```

Here's the correction I've applied:

```
if (itemsfound < 75):
    print('Too few results, increasing interval', end='')            
    diff = 2 - (itemsfound / 75)
    diff = min(MAXIMUM_EXPANSION_MULTIPLIER, diff)
    interval = int(interval * diff)
elif (itemsfound > 99):
    print('Too many results, reducing interval', end='')
    interval = int(interval * (0.8 - (0.05 * toomany_inarow)))
    toomany_inarow += 1
else:    
    lower = upper
    toomany_inarow = max(0, toomany_inarow-1)
    smartinsert(sql, cur, searchresults)

upper = lower + interval
```

**Comment by voussoir** [comment]
Hi @rroyo,

The comment on line [313](https://github.com/voussoir/reddit/blob/70829b786b1f53ee616eec384c1db5476d8034f0/Prawtimestamps/timesearch.py#L313) says "Intentionally not elif", because I did consider this scenario and I believe the current code is more effective.

Suppose a certain interval yields only 50 items. If we keep the lower bound the same and increase the upper bound, and we receive 90 items next time, the first 50 of those 90 are duplicates from the last request! By setting `lower = upper` **and** increasing the interval, we guarantee that the next request is going to contain fresh items, and we at least hope that this time it will be closer to the desired 75 to 99 items per request. Even if the next request still contains less than 75 items, at least they'll be 75 new ones and not 50 dupes plus 25.

The only time the lower bound should stay in place is when we receive 100 items in a request, because there's no way to know whether that 100th item is a perfect fit or there are more items in the requested range without doing pagination. In this case, I opted not to do any pagination, but use new search queries every time for simplicity. The code reflects this in that the lower bound moves as long as the `itemsfound > 99` check failed.

Hope that helps. My code comment should have been more descriptive about why I chose this.
